lib/spectests/tests/excludes.txt

```diff
@@ -266,6 +266,30 @@ clif:fail:data.wast:266:windows # AssertUnlinkable - caught panic Any
 clif:fail:data.wast:186:windows # AssertUnlinkable - caught panic Any
 clif:fail:data.wast:194:windows # AssertUnlinkable - caught panic Any
 
+# NaN canonicalization
+clif:fail:wasmer.wast:105 # AssertReturn - result I32(2143289345) ("0x7fc00001") does not match expected I32(2143289344) ("0x7fc00000")
```

Comment:
Any idea of why this happens? I thought Cranelift supported NaN canonicalization

Reply:
Looks like it defaults to off in cranelift and nothing turns it on?

https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/wasmerio/cranelift%24+enable_nan_canonicalization&patternType=literal
https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/wasmerio/wasmer%24+enable_nan_canonicalization&patternType=literal

Reply:
@losfair as part of this PR can you make Cranelift read the `nan_canonicalization` field from the Compiler `Config` and use it accordingly? Thanks!